ragen/env/deepcoder/env.py

```
@@ -0,0 +1,187 @@
+        else:
+            self.dataset = None
+
+    def _sample_problem(self, seed: Optional[int] = None) -> Tuple[str, str]:
+        if self.dataset is None:
+            prompt = "Write a function add(a, b) that returns a + b."
+            solution = "def add(a, b):\n    return a + b"
+            return prompt, solution
+
+        split = self.config.split or next(iter(self.dataset.keys()))
+        dataset_split = self.dataset[split]
+        index = random.randint(0, len(dataset_split) - 1)
```

Ruff (0.15.2)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)